app/contracts/base_contract.rb

```diff
@@ -102,16 +102,13 @@ def attribute_permission(attribute, permission)
 
     def add_writable(attribute, writable)
       attribute_name = attribute.to_s.delete_suffix("_id")
+      writable_conditions[attribute_name] = writable
 
-      unless writable == false
-        writable_attributes << attribute_name
-        # allow the _id variant as well
-        writable_attributes << "#{attribute_name}_id"
-      end
+      return if writable == false
 
-      if writable.respond_to?(:call)
-        writable_conditions << [attribute_name, writable]
-      end
```

Comment:
_Sherlock Bliard 🕵️_

Reply:
yep, not needed anymore as the writable value is always written to writable_conditions now (and it's a hash now too).